### ISSUE TITLE
add unit tests!!!

### ISSUE THREAD

**Issue by ExpandingMan** [ExpandingMan/Arrow.jl#11]
We need proper unit testing so that we can make changes and merge PR's properly.  I'll try to get to this right after I do 0.7 updates.

**Comment by ExpandingMan** [comment]
Have added some fairly extensive unit tests that mostly test whether `getindex` and various `ArrowVector` constructors are working properly.  Still need to add more, particularly for writing data.

**Comment by davidanthoff** [comment]
Just to second that, code coverage right now is sparse :)

https://codecov.io/gh/ExpandingMan/Arrow.jl

**Comment by ExpandingMan** [comment]
Yes, I've noticed that.  I might actually consider getting rid of some of the constructors, there are probably too many of them, I feel like half of this package is constructors.  The tests I've added so far were intended to make damn sure that this will never segfault (at least not because of Arrow, users can still cause it).  Fortunately 0.7 detects a large class of segfaults and throws a proper error for a large class of them at seemingly zero cost, so that's good.

Tests that we are currently missing that I am most concerned about are for writing.

**Comment by ExpandingMan** [comment]
Phew... ok I now have in a hell of a lot more unit tests.  I haven't seen what the coverage is yet, but now pretty much all of the "important" stuff is being tested.  Probably not going to do more myself for quite a while.
